airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -512,20 +514,25 @@ def request_params(
         stream_slice: Mapping[str, Any] = None,
         next_page_token: Mapping[str, Any] = None,
     ) -> MutableMapping[str, Any]:
-        next_page_token = next_page_token or {}
         parsed_state = self.check_stream_state(stream_state)
-        if self.cursor_field:
-            params = {"start_time": next_page_token.get(self.cursor_field, parsed_state)}
-        else:
-            params = {"start_time": calendar.timegm(pendulum.parse(self._start_date).utctimetuple())}
-        # check "start_time" is not in the future
-        params["start_time"] = self.check_start_time_param(params["start_time"])
+        params = {"start_time": self.check_start_time_param(parsed_state)}
         if self.sideload_param:
             params["include"] = self.sideload_param
         if next_page_token:
             params.update(next_page_token)
         return params
 
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        old_value = (current_stream_state or {}).get(self.cursor_field, 0)
+        new_value = (latest_record or {}).get(self.cursor_field, 0)
```

Comment:
Are we certain that records are transmitted in chronological order and there is no way to get records outside the current slice window? The past Salesforce bug has me cautious about this type of thing.

Is it valid for the stream_state and the latest_record to not have a `generated_timestamp` value. It doesn't seem right to save a timestamp of `0` as state. That would effectively turn this into an incredibly early sync on the next run (unless we get the max of start_date and state). Regardless, if this condition shouldn't be possible, then it feels more like we should be throwing some type of error

Reply:
updated with start date